cocos/scripting/js-bindings/manual/ScriptingCore.cpp

```diff
@@ -520,8 +520,8 @@ void ScriptingCore::string_report(JS::HandleValue val) {
 bool ScriptingCore::evalString(const char *string, JS::MutableHandleValue outVal, const char *filename, JSContext* cx, JS::HandleObject global)
 {
     JSAutoCompartment ac(cx, global);
-    JS::PersistentRootedScript script(cx);
-    if (script == nullptr) {
+    JS::RootedScript script(cx);
+    if (&script == nullptr) {
```

Comment:
@ricardoquesada I believe you're correct. I don't believe that if check is necessary. It seems likely that if check is a hold over from when we created a PersistentRootedObject on the heap here. It makes sense to do that check if we are doing a dynamic allocation, but since we are creating a Rooted<T> on the stack, I don't believe we need to check the value of operator& on the Rooted<T>. I don't believe we if check other Rooted<T>s we create on the stack in other places in code, so it makes sense to not have that if check here. 

w.r.t the PersistentRootedScript, In the issue I wrote 

> Also on a slightly related note, I am unsure on why we are using a JS::PersistentRootedScript here over a JS::RootedScript. The script in this use case has a definite lifetime, and is not cached or stored globally, so I see no point for using a PersistentRooted here.

The spidermonkey API suggests that PersistentRooted<T>s should be used for objects with dynamic lifetime that could change based on runtime parameters. Since the lifetime of the script object here is known at compile time, it seems more proper to use a JS::Rooted<T> instead of a PersistentRooted<T>.